### DISCUSSION TITLE
Why isn't there an OdbcOperator?

### DISCUSSION THREAD

**Discussion by dabla** [apache/airflow#32047] [General] [answered]
### Description

We would like to have an OdbcOperator so that we can easily execute SQL statements without the need of defining methods in our DAG using the OdbcHook.

### Use case/motivation

We would like to have an OdbcOperator, we're already writing our own one as a POC based on the BaseSQLOperator and also willing to contribute it to the Airflow project, unless there is an alternative or are there any plans to support OdbcHooks right off the SQLOperator?

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by pankajkoti** [comment, marked as answer]
hi @dabla,

There is an [airflow.providers.common.sql.operators.sql.SQLExecuteQueryOperator](https://github.com/apache/airflow/blob/79eac7687cf7c6bcaa4df2b8735efaad79a7fee2/airflow/providers/common/sql/operators/sql.py#L189) which extends the `BaseSQLOperator` and is the recommended way for running SQL queries.

Docs link: https://airflow.apache.org/docs/apache-airflow-providers-common-sql/stable/operators.html#execute-sql-query 

This I believe will allow you to run SQL queries for ODBC type connections too. Can you please use that and check if something is missing there?

**Comment by dabla** [comment]
Thank you for your swift answer :) I just found out the same thing while going through the source code (I saw the deprecation message in the JdbcOperator) stating the same you're saying.  Will try using the SQLExecuteQueryOperator and keep you posted.

**Comment by potiuk** [comment]
I think @pankajkoti is perfect answer. I converted this to a discussion where it can be continued.

Generelly note for the future, any issue that you start with a question mark should start as a GitHub discussion not an issue. When you open an issue you get informed about it clearly in the template (including a link to follow to create a new discussion entry). It would be great to read and follow those guidelines next time you have a question @dabla.

**Comment by potiuk** [comment]
And some comment why - we try to manage our issues to some extent. Issue is usually something that we can track, measure and see where we can improve the project, discussions are more for help/answering the questions and general discussion about the project.

**Comment by Joffreybvn** [comment]
Thanks again for your help !
With @dabla , we tested it and it works in almost all cases. We have however an issue with SELECT statement that [I detailed here](https://github.com/apache/airflow/discussions/32058), and that we are whilling to address.
